awesome_owl/static/src/counter/counter.js

```diff
@@ -0,0 +1,19 @@
+/** @odoo-module **/
+
+import { Component, useState } from "@odoo/owl";
+
+export class Counter extends Component {
+    static template = "awesome_owl.counter";
+    state = useState({ value: 0 });
+
+    static props = {
+        onChange: { type: Function, optional: true },
+    };
```

Comment:
```suggestion
    static props = {
        onChange: { type: Function, optional: true },
    };

    state = useState({ value: 0 });
```
I think it's better to have different blocks for class variables and instance variables.
I'm more accustomed to seeing the instance variable defined in the `setup` function even, but I don't mind this way to write it.